src/test/java/com/intel/gkl/testing_utils/CompressionDataProviders.java

```diff
@@ -0,0 +1,37 @@
+package com.intel.gkl.testing_utils;
+
+public class CompressionDataProviders {
+    @org.testng.annotations.DataProvider(name = "allCompressionCases")
+    public static Object[][] generateAllCompressionCases(){
+        int levels = 10;
+        Object[][] params = new Object[levels*2][];
```

Comment:
Object[][] params = new Object[levels*2][]; can be simply params int[10][2] , nowrap can take values 0 and 1 for false and true.

Reply:
Not necessarily. In TestNG DataProviders are required to return Object class. In this case, I want to have  2*levels arrays of 2 elements:
 1. {0, true}
 2. {1, true}
...
1 9. {8, false}
2 0. {9, false}

Each row will be injected into test function arguments. So the array initialization would look like `new Object[20][2]`. But I think it's more elegant to do it dynamically in a loop as I did.